packages/cli/src/api/getTranslationsForCatalog.test.ts

```diff
@@ -0,0 +1,337 @@
+    const actual = getTranslationsForCatalog(catalogStub, "ru", {
+      sourceLocale: "en",
+      fallbackLocales: {
+        default: "pl",
+      },
+      onMissing: missingSpy,
+    })
+
+    expect(missingSpy).not.toBeCalled()
+
+    expect(actual).toStrictEqual({
+      hashid1: "ru: translation: Lorem",
+      hashid2: "pl: translation: Ipsum",
+    })
+  })
+
+  it("Should fallback to single fallbackLocales", () => {
+    // prettier-ignore
+    const catalogStub = getCatalogStub({
+      ...lang("pl", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum")
+      ]),
+      ...lang("ru", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum", true)
+      ])
+    })
+
+    const missingSpy = jest.fn()
+    const actual = getTranslationsForCatalog(catalogStub, "ru", {
+      sourceLocale: "en",
+      fallbackLocales: {
+        ru: "pl",
+      },
+      onMissing: missingSpy,
+    })
+
+    expect(missingSpy).not.toBeCalled()
+
+    expect(actual).toStrictEqual({
+      hashid1: "ru: translation: Lorem",
+      hashid2: "pl: translation: Ipsum",
+    })
+  })
+
+  it("Should fallback to multiple fallbacks and then to default", () => {
+    // prettier-ignore
+    const catalogStub = getCatalogStub({
+      ...lang("pl", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum"),
+        message("hashid3", "Dolor")
+      ]),
+      ...lang("es", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum"),
+        message("hashid3", "Dolor", true),
+        message("hashid4", "Sit")
+      ]),
+      ...lang("de", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum", true),
+        message("hashid3", "Dolor", true)
+      ]),
+      ...lang("ru", [
+        message("hashid1", "Lorem"),
+        message("hashid2", "Ipsum", true),
+        message("hashid3", "Dolor", true),
+
+        // this is not present in first fallback locale DE, but present in next ES
+        message("hashid4", "Sit", true)
+      ])
+    })
+
+    const missingSpy = jest.fn()
+    const actual = getTranslationsForCatalog(catalogStub, "ru", {
+      sourceLocale: "en",
+      fallbackLocales: {
+        ru: ["de", "es"],
+        default: "pl",
+      },
+      onMissing: missingSpy,
+    })
+
+    expect(missingSpy).not.toBeCalled()
+    expect(actual).toStrictEqual({
+      hashid1: "ru: translation: Lorem",
+      // this fallback to de -> es
+      hashid2: "es: translation: Ipsum",
+      // this fallback to de -> es -> default
+
+      hashid3: "pl: translation: Dolor",
+      // this fallback to de -> es
+      hashid4: "es: translation: Sit",
+    })
+  })
+
+  it("Should take source messages when requested locale is sourceLocale", () => {
+    // prettier-ignore
+    const catalogStub = getCatalogStub({
+      ...lang("en", [
+        message("hashid1", "Lorem", true),
+        message("hashid2", "Ipsum", true)
+      ])
```

Comment:
I think you should add translated messages to the catalog to really test that `getTranslationsForCatalog` returns "**source messages** if requested locale is sourceLocale" as the test name says:

```suggestion
      ...lang("en", [
        message("hashid1", "Lorem", true),
        message("hashid2", "Ipsum", true)
      ]),
      
      ...lang("pl", [
        message("hashid1", "Lorem"),
        message("hashid2", "Ipsum")
      ])
```

Reply:
description is a bit misleading,  actually it doesn't matter for us would i take from source messages or translations. Because for source locale source == translation.  

Extractor always populates translations for sourceLocale. What is really matter is that when sourceLocale doesn't have a translation, we should fallback to the source message and don't call "missing" callback

i will change a description

Reply:
changed